contracts/MemeFactory.sol

```diff
@@ -578,16 +588,26 @@ abstract contract MemeFactory {
 
         uint256 amount = scheduledForAscendance;
         require(amount > 0, "Nothing to send");
+        require(_launched != 0 || (_launched == 0 && amount > launchAmountTarget), "Not enough to cover launch campaign");
```

Comment:
It's much simpler to handle the launch campaign in this method. Everything with the launch campaign is scoped to scheduledForAscendance, so handling it here makes the code much more readable